packages/api-client-core/src/FieldSelection.ts

```diff
@@ -3,5 +3,28 @@
  * Example: `{ id: true, name: false, richText: { markdown: true, html: false } }`
  **/
 export interface FieldSelection {
-  [key: string]: boolean | null | undefined | FieldSelection;
+  [key: string]: FieldSelectionType;
```

Comment:
In order to support `after` and `first` parameters we need to allow the user to pass through both strings and numbers values. Adding them directly to this type means that all fields would accept them which is not ideal. In order to mitigate this I have introduced a specific `ConnectionFieldSelection` that has the just pagination parameters and edges.